### PR TITLE
CORTX-27957: Add st for DIX repair using m0repair utility.

### PR DESCRIPTION
**Pull request by sanjognaik**
# Problem Statement
- Currently DIX repair STs use m0spiel utility for repair and rebalance, whereas SNS repair STs are use m0spiel as well as m0repair utilities. So new STs are added for DIX repair to use m0repair utility.

# Design
- Added enpoint for DIX CLI and DIX QUIESCE CLI
- Added Motr DIX repair ST using m0repair
- Added m0t1fs DIX repair ST using m0repair

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by shashank-parulekar — CHANGES_REQUESTED** [review]

File: dix/cm/st/m0t1fs_dix_repair_quiesce.sh [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+#
+
+M0_SRC_DIR=`readlink -f $0`
+M0_SRC_DIR=${M0_SRC_DIR%/*/*/*/*}
+
+. $M0_SRC_DIR/m0t1fs/linux_kernel/st/m0t1fs_common_inc.sh
+. $M0_SRC_DIR/m0t1fs/linux_kernel/st/m0t1fs_server_inc.sh
+. $M0_SRC_DIR/m0t1fs/linux_kernel/st/m0t1fs_dix_common_inc.sh
+. $M0_SRC_DIR/spiel/st/m0t1fs_spiel_dix_common_inc.sh # verify()
+
```
[the remark this answers is not in the document]
Comment (sanjognaik) [reply dix/cm/st/m0t1fs_dix_repair_quiesce.sh:28]: Already done.

File: dix/cm/st/dix_repair_st.sh [hunk 2]
```diff
@@ -95,48 +99,57 @@ device_query() {
...
 	local rc
-	spiel_wait_for_dix_repair
-	rc=$?
-	[ $rc != 0 ] && return $rc
+	if [ ${do_spiel} == 1 ]; then
+		spiel_wait_for_dix_repair || return $?
+	else
+		wait_for_dix_repair_or_rebalance "repair" || return $?
+	fi
 	cas_disk_state_set "repaired" $fail_device
```
[the remark this answers is not in the document]
Comment (sanjognaik) [reply dix/cm/st/dix_repair_st.sh]: Done.

File: dix/cm/st/dix_repair_st.sh [hunk 3]
```diff
@@ -95,48 +99,57 @@ device_query() {
...
 	local rc
-	spiel_wait_for_dix_rebalance
-	rc=$?
-	[ $rc != 0 ] && return $rc
+	if [ ${do_spiel} == 1 ]; then
+		spiel_wait_for_dix_rebalance || return $?
+	else
+		wait_for_dix_repair_or_rebalance "rebalance" || return $?
+	fi
 	cas_disk_state_set "online" $fail_device
```
[the remark this answers is not in the document]
Comment (sanjognaik) [reply dix/cm/st/dix_repair_st.sh]: Done.

File: dix/cm/st/m0t1fs_dix_repair.sh [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+	motr_service start $multiple_pools $stride $N $K $S $P || {
+		echo "Failed to start Motr Service."
+		return 1
+	}
+
+	for ((i=0; i < ${#IOSEP[*]}; i++)) ; do
+		ios_eps="$ios_eps -S ${lnet_nid}:${IOSEP[$i]}"
+	done
+
+	if [[ $rc -eq 0 ]] && ! dix_repair_test ; then
```
[the remark this answers is not in the document]
Comment (sanjognaik) [reply dix/cm/st/m0t1fs_dix_repair.sh]: Done.

**Review by nkommuri — APPROVED** [review]

**Review by shashank-parulekar — APPROVED** [review]

**Pull request merged**